### DISCUSSION TITLE
How to Add More Items?

### DISCUSSION THREAD

**Discussion by voloved** [kwsch/PKHeX#3706] [General]
Hi,

I'm working on a Pokemon Emerald hack and I added more items and moves.
Where can more items be added?
I see: `\PKHeX\PKHeX.Core\Resources\text\items\gen3\text_ItemsG3_en.txt`. Is that an enum?
Can I just add more items there and it knows they are the next item number, or is there more that I must change?

**Comment by sora10pls** [comment]
The file you are referencing is the English localization for item names in the Generation 3 games. Each line is a string that represents that index's item name. The same can be said for all the other text resources, such as [move strings](https://github.com/kwsch/PKHeX/blob/master/PKHeX.Core/Resources/text/other/en/text_Moves_en.txt).

If you plan on adding more _total_ items and moves than the original game, you can modify `Tables3.cs` to update the maximums defined there: https://github.com/kwsch/PKHeX/blob/b8f6b5f434218bf291fbcef211008f50cd6eee25/PKHeX.Core/Legality/Tables/Tables3.cs#L5-L15

**Comment by voloved** [comment]
Thank you so much!
I'll give it a shot. Also, are there any tutorials you'd point to for me to be able to run it?
I'm used to C and Python and after 30 minutes of Googling, and coming up short on why clicking Run in VSCode with the proper SDK does nothing.

**Comment by sora10pls** [comment]
The project's [readme](https://github.com/kwsch/PKHeX#readme) has instructions on how to build and run PKHeX.

**Comment by voloved** [comment]
Thanks . I did read through them, but could not find what commands specifically to use to run the program. When I tried to do a `dotnet run` after reading through other documents, I got the following message:

```
C:\Users\Chrono\Documents\GIT\Emerald\PKHeX\PKHeX.WinForms>dotnet run
C:\Program Files\dotnet\sdk\6.0.404\Microsoft.Common.CurrentVersion.targets(3262,5): error MSB3823: Non-string resource
s require the property GenerateResourceUsePreserializedResources to be set to true. [C:\Users\Chrono\Documents\GIT\Emer
ald\PKHeX\PKHeX.Drawing.PokeSprite\PKHeX.Drawing.PokeSprite.csproj]
C:\Program Files\dotnet\sdk\6.0.404\Microsoft.Common.CurrentVersion.targets(3262,5): error MSB3822: Non-string resource
s require the System.Resources.Extensions assembly at runtime, but it was not found in this project's references. [C:\U
sers\Chrono\Documents\GIT\Emerald\PKHeX\PKHeX.Drawing.PokeSprite\PKHeX.Drawing.PokeSprite.csproj]

The build failed. Fix the build errors and run again.
```

I am using .NET 6.0, which seems like it should work based on the tutorial, but I'm too novice in C# to be certain on what I should expect.
Thank for your help!

**Comment by sora10pls** [comment]
The .NET 6.0 build is experimental; you should install .NET Framework 4.6 and try building with an IDE like Visual Studio.

**Comment by voloved** [comment]
Got it. I will do what I can to install 4.6. It looks like Windows 10 doesn't make it easy since I have a later version of .NET framework installed on it by default and Windows doesn't allow an older version installed as well. I'll look into tracking down and uninstalling the newer version.
https://learn.microsoft.com/en-us/dotnet/framework/install/troubleshoot-blocked-installations-and-uninstallations?redirectedfrom=MSDN

**Comment by voloved** [comment]
Hey, I got it working. I was trying with VSCode and then tried Visual Studio and its installer instead. I'm unsure why, but installing v4.6 worked through the Visual Studio install, but not through the standard install file.
I'm mnetioning this in case anyone else runs into this issue as well.
